Documentation.app/Contents/MacOS/Composition.playground/Pages/Combining values of the same type.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,138 @@
+ /*:However, the operation is **not necessarily commutative**: combining `x` with `y` may not be the same as combining `y` and `x`. This becomes evident with Strings:
+ */
+let a = "Hello"
+let b = "World"
+
+a.combine(b) != b.combine(a)
+/*:
+ ## Having zero or more values
+ 
+ #### Problem
+ 
+ **I have**:
+ 
+ - Zero or more values of type `A`.
+ 
+ **I want to have**:
+ 
+ - A single value of type `A`.
+ 
+ #### Solution
+ 
+ The problem described here is similar to the problem above, with the difference of potentially having no values. We can use `combine`, but we need to provide a *default* value when no values are combined.
+ 
+ This default value is `empty` and it is available in the `Monoid` type class. `empty` has the property of being combinable with any other element of the type, yielding the same value it is combined with.
+ 
+ #### Example
+ 
+ Like with Semigroups, basic types in Swift already conform to `Monoid` in Bow.
+ */
+x.combine(Int.empty()) // Returns x
+Int.empty().combine(x) // Returns x
+/*:
+ ## Combining your own types
+ 
+ #### Problem
+ 
+ I have my own type and I need to combine elements of this type.
+ 
+ #### Solution
+ 
+ You need to use Swift `extension` mechanisms to make your type conform to `Semigroup` and `Monoid`, guaranteeing that your implementation honors the properties of both type classes.
+ 
+ #### Example
+ 
+ Consider a type to model a shopping cart:
+ */
+struct ShoppingCart<Product> {
+    let products: [Product]
+    let amount: Double
+}
+/*:
+ If I need to give it the ability to combine its values, I need to determine how each of its properties are combined with the ones in other values. In this case:
+ 
+ - `products` can be combined by concatenating the arrays of the two carts that are being combined.
+ - `amount` can be combined by adding the amounts of the two carts.
+ 
+ Therefore, I can make an `extension` to conform to `Semigroup`:
+ */
+extension ShoppingCart: Semigroup {
+    func combine(_ other: ShoppingCart<Product>) -> ShoppingCart<Product> {
+        ShoppingCart(
+            products: self.products + other.products,
+            amount: self.amount + other.amount
+        )
+    }
+}
+/*:
+ If I need to give it the ability to have an empty value, I need to determine how each of its properties need to be initialized to an empty value. In this case:
+ 
+ - `products` can be initialized to an empty array.
+ - `amount` can be initialized to `0.0`.
+ 
+ Therefore, I can make an `extension to conform to `Monoid`:
+ */
+extension ShoppingCart: Monoid {
+    static func empty() -> ShoppingCart<Product> {
+        ShoppingCart(
+            products: [],
+            amount: 0.0
+        )
+    }
+}
+/*:
+ ## Are all types that conform to Semigroup also conform to Monoid?
+ 
+ If your type conforms to `Monoid` it also must conform to `Semigroup`, but the reverse is not true. You may be able to combine values of a type, but not be able to provide an empty value.
```

Comment:
It will be nicer if we could embed this kind of paragraphs Q/A in "hidden" (collapsed) UI elements. - @calvellido ?

Reply:
Is there anything like that in standard Markdown? In this case, it is just this section, so I think it is not totally mandatory.